src/Administration/Resources/app/administration/src/module/sw-cms/component/sw-cms-reset-inheritance/index.ts

```diff
@@ -0,0 +1,57 @@
+import template from './sw-cms-reset-inheritance.html.twig';
+import './sw-cms-reset-inheritance.scss';
+
+const { set, merge } = Shopware.Utils.object;
+
+/**
+ * @private
+ * @sw-package discovery
+ */
+export default Shopware.Component.wrapComponentConfig({
+    template,
+    mixins: [
+        Shopware.Mixin.getByName('cms-state'),
+    ],
+    data() {
+        return {
+            showModal: false,
+        }
+    },
+    computed: {
+        cmsPageStore() {
+            return Shopware.Store.get('cmsPage');
+        },
+        hasOverrides() {
+            return !Shopware.Utils.types.isEmpty(this.contentEntity?.slotConfig);
+        },
+    },
+    methods: {
+        async onConfirm() {
+            this.showModal = false;
+
+            this.resetSlotOverrides();
+
+            /**
+             * Run watchers before removing the slotConfig to ensure sw-cms-form-sync won't
+             * override the reset.
+             */
+            await this.$nextTick();
+
+            set(this.contentEntity!, 'slotConfig', null);
+        },
+        resetSlotOverrides() {
+            this.cmsPageStore.currentPage?.sections?.forEach((section) => {
+                section.blocks?.forEach((block) => {
+                    block.slots?.forEach((slot) => {
+                        if (!slot.config) {
+                            return;
+                        }
+
+                        const origin = slot.getOrigin();
+                        slot.config = merge(slot.config, origin.translated?.config ?? origin.config ?? {});
```

Comment:
```suggestion
                        merge(slot.config, origin.translated?.config ?? origin.config ?? {});
```
This method mutates object. https://lodash.com/docs/4.17.15#merge